spec/unit/platform/query_helpers_spec.rb

```diff
@@ -114,6 +114,26 @@
   end
 end
 
+describe "Chef::Platform#supports_msi?" do
+  let(:key) { "System\\CurrentControlSet\\Services\\msiserver" }
+  let(:key_query_value) { 0x0001 }
+  let(:access) { key_query_value }
+  let(:error_code) { 0 }
+
+  it "returns false early when not on windows" do
+    allow(ChefConfig).to receive(:windows?).and_return(false)
+    expect(Chef::Platform.supports_msi?).to be false
+  end
+
+  it "returns true when the registry key exists" do
+    allow(ChefConfig).to receive(:windows?).and_return(true)
+    allow(Win32::Registry::HKEY_LOCAL_MACHINE).to receive(:open).
```

Comment:
You'll need to define `Win32::Registry::HKEY_LOCAL_MACHINE`. Because `:open` is a private method for some objects, I chose to use a double [defined here](https://github.com/chef/chef/blob/master/spec/unit/platform/query_helpers_spec.rb#L38) and [used here](https://github.com/chef/chef/blob/master/spec/unit/platform/query_helpers_spec.rb#L56)